R/delete_current_year_model_runs.R

```diff
@@ -0,0 +1,83 @@
+# Script to delete a list of model runs by ID from AWS.
+#
+# Accepts one argument, a comma-delimited list of run IDs for model runs
+# whose artifacts should be deleted.
+#
+# Assumes that model runs are restricted to the current assessment cycle, where
+# each assessment cycle starts in April. Raises an error if no objects matching
+# a given ID for the current year could be located in S3. This error will get
+# raised before any deletion occurs, so if one or more IDs are invalid then
+# no objects will be deleted.
+#
+# Example usage (delete the runs 123, 456, and 789 in the current year):
+#
+#   delete_current_year_model_runs.R 123,456,789
+
+library(glue)
+library(here)
+library(magrittr)
+source(here("R", "helpers.R"))
+
+current_date <- as.POSIXct(Sys.Date())
+current_month <- current_date %>% format("%m")
+current_year <- current_date %>% format("%Y")
+
+# The following heuristic determines the current upcoming assessment cycle year:
+#
+#   * From April to December (post assessment), `year` = next year
+#   * From January to March (during assessment), `year` = current year
+year <- if (current_month < "03") {
+  current_year
+} else {
+  as.character(as.numeric(current_year) + 1)
+}
+
+# Convert the comma-delimited input to a vector of run IDs. Accepting one or
+# more positional arguments would be a cleaner UX, but since this script is
+# intended to be called from a dispatched GitHub workflow, it's easier to parse
+# one comma-delimited string than split a space-separated string passed as a
+# workflow input
```

Comment:
I agonized over this interface, perhaps unnecessarily; I'm curious if you agree that a single comma-delimited string is the right compromise!

Reply:
I think it's fine! Though we _might_ want to use a space-delimited string in order to match [build-and-test-dbt](https://github.com/ccao-data/data-architecture/actions/workflows/build_and_test_dbt.yaml). It would have to be quoted i.e. not positional args. I'm really fine with either though as long as it's clear what the input is supposed to be.

Reply:
Cool, I switched the workflow input to accept a space-delimited string in a4895c20d2b44cad197df855ac7cdbeebbbe9616.